gms/inet_address.hh

```diff
@@ -58,9 +58,9 @@ public:
     bytes_view bytes() const noexcept {
         return bytes_view(reinterpret_cast<const int8_t*>(_addr.data()), _addr.size());
     }
-    friend inline bool operator==(const inet_address& x, const inet_address& y) noexcept = default;
-    friend inline bool operator<(const inet_address& x, const inet_address& y) noexcept {
-        return x.bytes() < y.bytes();
+    bool operator==(const inet_address&) const noexcept = default;
+    bool operator<(const inet_address& y) const noexcept {
+        return bytes() < y.bytes();
     }
```

Comment:
What about operator<=>?

Reply:
Can do.
Can it affect performance if it is used to derive the default `operator<`?
Or should we have both?

Reply:
Hmm
```
using bytes_view = std::basic_string_view<int8_t>;
```

```
#if __cpp_lib_three_way_comparison
  template<typename _CharT, typename _Traits>
    [[nodiscard]]
    constexpr bool
    operator==(basic_string_view<_CharT, _Traits> __x,
	       type_identity_t<basic_string_view<_CharT, _Traits>> __y)
    noexcept
    { return __x.size() == __y.size() && __x.compare(__y) == 0; }

  template<typename _CharT, typename _Traits>
    [[nodiscard]]
    constexpr auto
    operator<=>(basic_string_view<_CharT, _Traits> __x,
		__type_identity_t<basic_string_view<_CharT, _Traits>> __y)
    noexcept
    -> decltype(__detail::__char_traits_cmp_cat<_Traits>(0))
    { return __detail::__char_traits_cmp_cat<_Traits>(__x.compare(__y)); }
#else
```

So we already use `operator<=>` under the covers.
Will change accordingly.

Reply:
Sometimes it's better to provide both operator== and operator<=>, if operator== can be significantly faster.